### DISCUSSION TITLE
Labels are not updated when MR does

### DISCUSSION THREAD

**Discussion by dor272** [renovatebot/renovate#22485] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v34.160.0 (renovate-runner v10.186.0)

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Hi,
I encountered a problem where renovate originally opened an MR to update a package patch.
I didn't merge it, time went by, and the package released a new minor release.
Renovate detected it, and updated the MR title and description, not the labels.

These are the labels I set in my config:
``` 
 "major": {
    "addLabels": ["Dependencies::major"],
    "stabilityDays": 60,
    "prConcurrentLimit": 3
  },
  "minor": {
    "addLabels": ["Dependencies::minor"]
  }
```

It seems like package updates are detected, and work, but label rules are not considered once it is changed on an existing MR.
I can't recover the logs since it happened a few weeks ago, and it did change the title, but I had to fix the label manually.

![Screenshot from 2023-05-29 17-29-13](https://github.com/renovatebot/renovate/assets/26774672/da70f12a-cdf6-4339-b2c5-e65fb8fa3d83)

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment, marked as answer]
This is by design. From the documentation:

![image](https://github.com/renovatebot/renovate/assets/6311784/0bd93997-43b2-477f-b6e2-4c46f642909e)

Your use case could be a third dot point added there.

The reason Renovate doesn't "reapply" labels to PRs is because it would be hard to do so without fighting with users who want to remove them.

It would be a hard problem to solve, but not impossible

**Reply by dor272** [reply]
So is there a way to restrict renovate from "promoting" an existing PR from patch to minor? That would solve my problem as well.

**Reply by rarkins** [reply]
Not that I can think of.

I have created #22496 as a feature request issue

**Reply by viceice** [reply]
> So is there a way to restrict renovate from "promoting" an existing PR from patch to minor? That would solve my problem as well.

https://docs.renovatebot.com/configuration-options/#separateminorpatch
